apollo-federation/src/operation/mod.rs

```diff
@@ -3045,9 +3045,7 @@ impl SelectionSet {
                     path: Vec::new(),
                     field: field.clone(),
                 }),
-                Selection::FragmentSpread(_fragment) => {
-                    todo!()
-                }
+                Selection::FragmentSpread(_fragment) => return fields,
```

Comment:
@duckki I could use your input on this one, I know you worked with this code just recently. Is the panic here (`todo!()` or `unreachable!()`) absolutely necessary here? Can we return what would be an empty vec with `return fields`? Or will that mess things up down the line?

Reply:
Yes, returning an empty vec is sensible here. But, please add an debug assertion so we can catch it during testing.
BTW, I'd like to move it into `optimize.rs`, where it should belong. But, we can do that refactoring later.

Reply:
@duckki , should we return `vec![]` or `fields`?

Reply:
Oh, I meant "not add to `fields` and keep going on".

Reply:
```suggestion
                Selection::FragmentSpread(_fragment) => {
                    debug_assert!(false, "unexpected fragment spreads in expanded fetch operation");
                }
```